src/main/java/org/kcctl/command/DescribePluginCommand.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright 2021 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.kcctl.command;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.kcctl.service.ConfigInfos;
+import org.kcctl.service.KafkaConnectApi;
+import org.kcctl.util.ConfigurationContext;
+import org.kcctl.util.Version;
+
+import com.github.freva.asciitable.AsciiTable;
+import com.github.freva.asciitable.Column;
+import com.github.freva.asciitable.HorizontalAlign;
+
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+@Command(name = "plugin", description = "Displays information about given plugin")
+public class DescribePluginCommand implements Callable<Integer> {
+
+    @CommandLine.Spec
+    CommandLine.Model.CommandSpec spec;
+
+    @CommandLine.Parameters(paramLabel = "PLUGIN NAME", description = "Name of the plugin")
```

Comment:
Let's implement a tab completion of the name; it will be particularly handy as the plug-in names are so long typically. Unlike "static" completions for option names, this requires to invoke kcctl from the completion script, retrieving the available plug-ins. You can find an example [here](https://github.com/kcctl/kcctl/blob/main/src/main/java/org/kcctl/command/DescribeConnectorCommand.java#L53) (essentially, that custom completion retrieves the names by calling another command which is hidden from the help text and completion itself).

Then regenerate the actual completion script itself, as described [here](https://github.com/kcctl/kcctl#updating-the-completion-script) (note the special backtick handling for that additional new _kcctl_ invocation). This will also automatically add the completion handling for the new types option and its allowed values.

Reply:
Do we need to do this in addition of using an enum? I guess this way it would only show types of plugins installed instead of the full list of types but I'm not sure this adds a lot of value.

Reply:
I was thinking of a completion of the plug-in _name_, not type, for the describe command; e.g. like so:

```
kcctl describe plugin io.debezium<TAB>
io.debezium.postgres.PostgresConnector...
```

That way, you don't have to type out the FQNs for the describe command. I have pushed a commit for adding that. Perhaps you can give it a spin and let me know whether that completion works for you as it should?